### DISCUSSION TITLE
Reading hard coded files/paths relative to an example or executable

### DISCUSSION THREAD

**Discussion by gnikit** [fortran-lang/fpm#742] [General]
I have an fpm module and in its examples I have certain demos that require for hard coded input files to be read in, with paths relative to the source file. However running `fpm run --example demo` runs from within `build/.../example/..`. Is there a way to:

1. Copy the input files manually?
2. Run a setup script before the example executes?
3. Change the running directory of the example?

I think any of the above would resolve the issue, the best solution in my opinion would be to optionally allow execution of setup and teardown scripts.

**Comment by LKedward** [comment]
Option 3 would be very useful to have; see issue #410 for the discussion.

**Reply by gnikit** [reply]
Thanks @LKedward I've gone ahead an upvoted the issue, currently the only working solution I've found is:

```sh
fpm run --example t13 --runner cp -- example/fortran/ && pushd example/fortran/ && ./t13 && pop
```

Not ideal but it's better than not working at all
